Runtime/Scripts/PersistentConnection.cs

```diff
@@ -131,192 +218,176 @@ IEnumerator OnIntervalEdgeEventsLocation(EdgeEventsConnection connection)
-              integration.NewFindCloudletHandler(EdgeEventsStatus.error, fcEvent);
-              break;
-            default:
-            case EventUnknown:
-              Logger.Log("Received Unknown Event");
-              break;
+            PropagateError(FindCloudletEventTrigger.LatencyTooHigh,
+              "New Cloudlet obtained from FindCloudletPerformanceMode is the same as old cloudlet");
           }
         }
+        else
+        {
+          PropagateError(FindCloudletEventTrigger.LatencyTooHigh, "FindCloudletPerformanceMode failed");
+        }
+      }
+    }
 
-    private void OnApplicationFocus(bool focus)
+    async void CompareLatencies(Statistics receivedStats)
     {
-      if (locationUpdatesRunning)
+      AppPort appPort;
+      Site site;
+      var netTest = new NetTest(integration.matchingEngine);
+      if (hasTCPPorts)
       {
-        StartCoroutine(OnIntervalEdgeEventsLocation(integration.matchingEngine.EdgeEventsConnection));
+        appPort = integration.GetAppPort(LProto.Tcp, config.latencyTestPort == 0 ? 0 : config.latencyTestPort);
+        site = new Site { host = appPort.FqdnPrefix + integration.latestFindCloudletReply.Fqdn, port = appPort.PublicPort, testType = TestType.CONNECT };
       }
-      if (latencyUpdatesRunning)
+      else
       {
-        StartCoroutine(OnIntervalEdgeEventsLatency(integration.matchingEngine.EdgeEventsConnection, integration.GetHost()));
+        appPort = integration.GetAppPort(LProto.Udp);
+        site = new Site { host = appPort.FqdnPrefix + integration.latestFindCloudletReply.Fqdn, port = appPort.PublicPort, testType = TestType.PING };
+      }
+      await netTest.TestSite(site);
+      double normalizedLatency = receivedStats.Avg - (receivedStats.Avg * config.performanceSwitchMargin);
+      if (site.average < normalizedLatency)
+      {
+        FindCloudletEvent findCloudletEvent = new FindCloudletEvent();
+        findCloudletEvent.trigger = FindCloudletEventTrigger.LatencyTooHigh;
+        findCloudletEvent.newCloudlet = integration.latestFindCloudletReply;
+        EdgeEventsStatus eventStatus = new EdgeEventsStatus(Status.success);
+        integration.NewFindCloudletHandler(eventStatus, findCloudletEvent);
+      }
+      else
+      {
+        PropagateError(FindCloudletEventTrigger.LatencyTooHigh,
+          "Cloudlet obtained from FindCloudletPerformanceMode have higher latency than the previous cloudlet");
       }
     }
 
-    private void OnApplicationPause(bool pause)
-    {
-      StopAllCoroutines();// stop edge events streaming
-    }
-
-    private void OnApplicationQuit()
+    void CleanUp()
     {
-      StopAllCoroutines();
+      latencyUpdatesCounter = 0;
+      latencyUpdatesRunning = false;
+      locationUpdatesCounter = 0;
+      locationUpdatesRunning = false;
+      location = null;
     }
 
-    private void OnDestroy()
+    void PropagateError(FindCloudletEventTrigger trigger, string error_msg)
```

Comment:
Is there a better name? Every error a FindCloudletEventTrigger.

Reply:
Maybe I can update the name since you don't like it (I am curious why? Is not representative of the method functionality!) 
Would something PropagateEdgeEventError be more coherent with the context?

Reply:
Given there's one for now, perhaps we can just rely on method name overloading for now to distinguish. Fine as is then.